src/Common/tests/System/AssertExtensions.cs

```diff
@@ -29,13 +29,26 @@ public static void Throws<T>(string netCoreParamName, string netFxParamName, Act
                 Assert.Equal(expectedParamName, exception.ParamName);
         }
 
-        public static void Throws<T>(string paramName, Action action)
+        public static T Throws<T>(string paramName, Action action)
             where T : ArgumentException
         {
             T exception = Assert.Throws<T>(action);
 
             if (!RuntimeInformation.FrameworkDescription.StartsWith(".NET Native"))
                 Assert.Equal(paramName, exception.ParamName);
+
+            return exception;
+        }
+
+        public static T Throws<T>(string paramName, Func<object> testCode)
+            where T : ArgumentException
+        {
+            T exception = Assert.Throws<T>(testCode);
+
+            if (!RuntimeInformation.FrameworkDescription.StartsWith(".NET Native"))
```

Comment:
@ianhays, didn't you submit a PR to a while ago to remove a class (`RuntimeDetection`) that made this less typo-prone?